kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSTypeImpl.kt

```diff
@@ -110,12 +110,12 @@ class KSTypeImpl private constructor(internal val type: KtType) : KSType {
     }
 
     override fun replace(arguments: List<KSTypeArgument>): KSType {
-        return type.replace(arguments.map { it.toKtTypeProjection() })?.let { getCached(it) } ?: KSErrorType
+        return type.replace(arguments.map { it.toKtTypeProjection() })?.let { getCached(it) } ?: KSErrorType()
     }
 
     override fun starProjection(): KSType {
         return type.replace(List(type.typeArguments().size) { KtStarTypeProjection(type.token) })
-            ?.let { getCached(it) } ?: KSErrorType
+            ?.let { getCached(it) } ?: KSErrorType()
```

Comment:
should we add something to replace type arguments use case? This is not helping with error types.

Reply:
I've tried to generalize the error type creation in such cases and introduced an `errorTypeOnInconsistentArguments` utility function which is now used in the old and AA impls. This pretty much covers all unfilled error types, except some where there's really nothing meaningful to report.